### ISSUE TITLE
problem with dependencies and --try-software-version in easybuild 2.0.0

### ISSUE THREAD

**Issue by pescobar** [hpcugent/easybuild-framework#1221]
I am trying this with EasyBuild 2.0.0 and I get an error:

```
[soft@vhircluster easybuild]$ eb --configfiles=easybuild.cfg /tmp/snpEff-4.0-Java-1.7.0_67.eb --try-software-version=4.1 --debug
== temporary log file in case of crash /tmp/eb-Nhg_RU/easybuild-KL98hT.log
ERROR: EasyBuild crashed with an error (at easybuild/tools/robot.py:232 in resolve_dependencies): Irresolvable dependencies encountered: Java/1.7.0_67
```

But I have this module installed:

```
[soft@vhircluster easybuild]$ ml av Java/1.7.0_67

-------------------- /home/soft/modules/all ------------------------------
   Java/1.7.0_67
```

This is the debug log:
https://github.com/statgen/bamUtil/archive/v1.0.13.tar.gz

**Comment by boegel** [comment]
Hmm, you don't have `--robot` enabled yourself in anyway here?

**Comment by fgeorgatos** [comment]
yeah, I think I know what the problem here is, it indeed surprises us the
first time:
- in effect, easybuild relies on *both* the module file and the easyconfig
file to sanely resolve its dependencies.

On 15 March 2015 at 18:23, Kenneth Hoste <notifications@github.com> wrote:

> Hmm, you don't have --robot enabled yourself in anyway here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-framework/issues/1221#issuecomment-81188194>
> .
>



-- 


echo "sysadmin know better bash than english"|sed s/min/mins/ \
  | sed 's/better bash/bash better/' # signal detected in a CERN forum

**Comment by pescobar** [comment]
@fgeorgatos I know EB relies in the easyconfigs to resolve dependencies when doing --dry-run but this was not the case for EB 1.x

@boegel this changed for EB 2.x ?

**Comment by boegel** [comment]
Use of `--try-X` implies `--robot`, but it probably shouldn't in this particular case, since `--try-software-version` basically implies *not* resolving dependencies (since adjusting the software version across the whole dep graph doesn't make sense).
